File: apps/api/app/schemas/trust.py
"""Trust scoring schemas — response models for trust and risk assessment.

FM-050: Pydantic models for trust scores.
"""

import uuid
from datetime import datetime

from pydantic import BaseModel

from app.models.trust_score import RiskLevel, EntityType


class TrustScoreRead(BaseModel):
    id: uuid.UUID
    entity_type: EntityType
    entity_id: uuid.UUID
    trust_score: float
    confidence: float
    risk_level: RiskLevel
    factors: dict | None
    project_id: uuid.UUID | None
    run_id: uuid.UUID | None
    assessed_at: datetime

    model_config = {"from_attributes": True}


class TrustScoreList(BaseModel):
    items: list[TrustScoreRead]
    total: int

File: apps/web/types/trust.ts
/** Trust scoring types matching the backend schemas (FM-046/FM-050). */

export type RiskLevel = "low" | "medium" | "high" | "critical";
export type EntityType = "task" | "artifact" | "run";

export interface TrustScore {
  id: string;
  entity_type: EntityType;
  entity_id: string;
  trust_score: number;
  confidence: number;
  risk_level: RiskLevel;
  factors: Record<string, unknown> | null;
  project_id: string | null;
  run_id: string | null;
  assessed_at: string;
}

export interface TrustScoreList {
  items: TrustScore[];
  total: number;
}

export interface RiskSummary {
  run_id: string;
  overall_risk: RiskLevel;
  total_assessments: number;
  risk_breakdown: Record<string, number>;
  high_risk_tasks: string[];
}
